controller/composite/controller.go

```diff
@@ -562,15 +561,15 @@ func (pc *parentController) claimChildren(parent *unstructured.Unstructured) (co
 func (pc *parentController) updateParentStatus(parent *unstructured.Unstructured, status map[string]interface{}) (*unstructured.Unstructured, error) {
 	// Overwrite .status field of parent object without touching other parts.
 	// We can't use Patch() because we need to ensure that the UID matches.
-	// TODO(enisoc): Use /status subresource when that exists.
-	// TODO(enisoc): Update status.observedGeneration when spec.generation starts working.
-	return pc.parentClient.Namespace(parent.GetNamespace()).AtomicUpdate(parent, func(obj *unstructured.Unstructured) bool {
+	return pc.parentClient.Namespace(parent.GetNamespace()).AtomicStatusUpdate(parent, func(obj *unstructured.Unstructured) bool {
 		oldStatus := k8s.GetNestedField(obj.UnstructuredContent(), "status")
 		if reflect.DeepEqual(oldStatus, status) {
 			// Nothing to do.
 			return false
 		}
+
 		k8s.SetNestedField(obj.UnstructuredContent(), status, "status")
+		k8s.SetNestedField(obj.UnstructuredContent(), parent.GetGeneration(), "status", "observedGeneration")
```

Comment:
Sorry for not noticing earlier, but we should add a check in at least one of the tests that status.observedGeneration is getting set correctly.

Reply:
In the long run, I suggest adding integration tests.